### PR TITLE
[SPARK-33441][BUILD][FOLLOWUP] Make unused-imports check for SBT specific.

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Move "unused-imports" check config to `SparkBuild.scala` and make it SBT specific.

### Why are the changes needed?
Make unused-imports check for SBT specific.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass the Jenkins or GitHub Action

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -27,6 +27,7 @@ import scala.collection.JavaConverters._
 import scala.collection.Map
 import scala.collection.immutable
 import scala.collection.mutable.HashMap
+import scala.collection.mutable.HashSet
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/SparkContext.scala:30]: It's for testing right?

**Review by HyukjinKwon — APPROVED** [review]
Looks good, pending tests.

File: core/src/main/scala/org/apache/spark/SparkContext.scala (hunk 1 above)
Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/SparkContext.scala:30]: Yeah

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/SparkContext.scala:30]: Expect this line to trigger compile error

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/SparkContext.scala:30]: image
Looks like it's working in Scala 2.12

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/SparkContext.scala:30]: image
`-Wunused:imports` working in Scala 2.13

**Comment by LuciferYang** [comment]
@HyukjinKwon done ~ waiting tests

File: graphx/src/main/scala/org/apache/spark/graphx/impl/VertexPartitionBase.scala [hunk 2]
```diff
@@ -17,7 +17,6 @@
 
 package org.apache.spark.graphx.impl
 
-import scala.language.higherKinds
```
Comment (LuciferYang) [inline graphx/src/main/scala/org/apache/spark/graphx/impl/VertexPartitionBase.scala:20]: checked by Scala 2.13 compiler

File: graphx/src/main/scala/org/apache/spark/graphx/impl/VertexPartitionBaseOps.scala [hunk 3]
```diff
@@ -17,7 +17,6 @@
 
 package org.apache.spark.graphx.impl
 
-import scala.language.higherKinds
```
Comment (LuciferYang) [inline graphx/src/main/scala/org/apache/spark/graphx/impl/VertexPartitionBaseOps.scala:20]: checked by Scala 2.13 compiler

File: project/SparkBuild.scala [hunk 4]
```diff
@@ -230,6 +231,8 @@ object SparkBuild extends PomBuild {
           // see `scalac -Wconf:help` for details
           "-Wconf:cat=deprecation:wv,any:e",
           // 2.13-specific warning hits to be muted (as narrowly as possible) and addressed separately
+          // TODO Enable this option when Scala 2.12 is no longer supported.
```
Comment (LuciferYang) [inline project/SparkBuild.scala:234]: @HyukjinKwon  A little interesting:

image

Scala 2.13 think this is a `unused import` but Scala 2.12 not. so comments `-Wunused:imports` in Scala 2.13 and left a TODO

Comment (HyukjinKwon) [reply project/SparkBuild.scala:234]: Can you file a JIRA under the Scala 2.13 umbrella JIRA? You can make a comment like `TODO(SPARK-XXXXX): blah blah`.

Comment (LuciferYang) [reply project/SparkBuild.scala:234]: OK

Comment (LuciferYang) [reply project/SparkBuild.scala]: Address 911acfe fix this

**Comment by HyukjinKwon** [comment]
Merged to master. Thanks for working on this @LuciferYang.

**Pull request closed without merging**